### PR TITLE
Add integration test for event entpoint

### PR DESCRIPTION
**Pull request by DianaJacota**
1. Refactor the project into a library and a binary: all the logic will live in the library crate while the binary itself will be just an entry point with a very slim main function.
This is needed in order to be able to add a new tests folder for our integration tests.

Anything under tests folder ends up being compiled in its own binary - all our code under test is imported as a crate. But the project, at the moment, is a binary: it is meant to be executed, not to be shared. Therefore, we can’t import the main function in our tests as it is right now

More details in chapter 3.4. OUR FIRST INTEGRATION TEST of "Zero to production" by Luca Palmieri

2. Add integration test for POST /event endpoint

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,4 +22,7 @@ actix-web = { version = "4.4.0", features = ["rustls-0_21"]}
 rustls = "0.21.7"
 rustls-pemfile = "1.0.3"
 actix-cors = "0.6.4"
+reqwest = { version = "0.11", features = ["json"] }
+tokio = { version = "1", features = ["macros", "rt-multi-thread"] }
```
Comment (aguzovatii) [inline Cargo.toml:26]: I suggest to check if we need both features, especially `rt-multi-thread`
Also from what I found, there is an alternative crate, called `tokio-test`

File: tests/event.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    // Assert
+    assert!(response.status().is_success());
+    assert_eq!(Some(0), response.content_length());
+}
+
+// launch the server as a background task
+fn spawn_app() {
+    let server = run().expect("Failed to bind address");
+    let _ = tokio::spawn(server);
```
Comment (aguzovatii) [inline tests/event.rs:32]: Is this running instance of `calendar-backend` killed at the end of the test?

**Review by aguzovatii — APPROVED** [review]

**Comment by aguzovatii** [comment]
Looks really great!

File: tests/event.rs (hunk 2 above)
Comment (DianaJacota) [reply tests/event.rs:32]: In [this documentation](https://docs.rs/tokio/latest/tokio/task/fn.spawn.html) is specified that:

> When a runtime is shutdown, all outstanding tasks are dropped, regardless of the lifecycle of that task.

Not the most accurate source, but ChatGPT says:

> Here's how multiple tests with #[tokio::test] work:
> 
> 1. **Sequential Execution**: The test functions annotated with #[tokio::test] within the same module will be executed one at a time, in the order they are defined in the code.
> 
> 2. **Shared Tokio Runtime**: All of these tests will share the same instance of the Tokio asynchronous runtime. The Tokio runtime is set up once before the first test runs and is torn down once after the last test completes.
> 
> 3. **Isolation**: Each test is isolated from the others in terms of state and resource management. Any changes made to the state or resources within one test should not affect the others.
> 
> 4. **Cleanup**: The Tokio runtime is responsible for cleaning up any asynchronous tasks, resources, and other runtime-specific components after all tests have completed.

File: Cargo.toml (hunk 1 above)
Comment (DianaJacota) [reply Cargo.toml]: I removed the in commit [e8b1589a83bbd1d77dd250e34180fb33c02241d2](https://github.com/calendar-team/calendar-backend/pull/11/commits/e8b1589a83bbd1d77dd250e34180fb33c02241d2#diff-117f5e52f7246f4eff31874f46a65d966e1bab790eacb121dd7e857a33c4974f)
Nice catch

`tokio-test` doesn't have an alternative for the spawn function so we can't use it easily as an alternative

**Pull request merged**